troposphere/static/js/components/common/ProjectCreateView.react.js

```diff
@@ -55,15 +93,12 @@ export default React.createClass({
         let descriptionErrorMessage = null;
 
         if (this.state.showValidation) {
-            if (projectName === "") {
-                nameClassNames = "form-group has-error";
-                nameErrorMessage = "This field is required";
-            }
-
-            if (projectDescription === "") {
-                descriptionClassNames = "form-group has-error";
-                descriptionErrorMessage = "This field is required";
-            }
+            nameClassNames = this.validateName().hasError ?
+                "form-group has-error" : null;
+            nameErrorMessage = this.validateName().message;
+            descriptionClassNames = this.validateDescription().hasError ?
+                "form-group has-error" : null;
```

Comment:
We could this potentially uses ['classnames'](https://github.com/JedWatson/classnames): 

```
nameClassNames = classnames({
    'form-group': this.validateName().hasError, 
   'has-error': this.validateName().hasError})
```

So we have a redundant boolean, but remove the ternary operator. I'm not sure which I favor at this point - so that's why I didn't say anything before merging (I could go either way right now).